LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -121,7 +175,11 @@ List<Module> GetModuleList()
             int minModuleCountLimit = 1;
             maxModuleCountLimit = Math.Min(totalModuleCount, maxModuleCountLimit);
             minModuleCountLimit = Math.Min(totalModuleCount, minModuleCountLimit);
-            int count = AttributeGenerator.random.Next(minModuleCountLimit, maxModuleCountLimit + 1);
+            int count = AttributeGenerator.random.Next(minModuleCountLimit, maxModuleCountLimit);
+            maxModuleCountLimit = Math.Min(totalModuleCount, maxModuleCountLimit);
+            minModuleCountLimit = Math.Min(totalModuleCount, minModuleCountLimit);
+            int totalmoduleCount = AttributeGenerator.random.Next(minModuleCountLimit, maxModuleCountLimit + 1);
```

Comment:
Why did you declare totalmoduleCount? You don't use it.  
On line 173 you declared totalModuleCount to get the number of all modules generated so far.
On line 178 you declared the count variable that you use do decide how many modules you want to add to the module list returned by GetModuleList(). 
You don't need totalmoduleCount on line 181, please remove.

Reply:
The whole logic is incorrect and very messy. Why do you call this two lines twice?
maxModuleCountLimit = Math.Min(totalModuleCount, maxModuleCountLimit);
minModuleCountLimit = Math.Min(totalModuleCount, minModuleCountLimit);

Why do you need totalmoduleCount on line 181?

Reply:
DONE

Reply:
DONE